### PR TITLE
Add a `PathConvertible` for `URI`

### PR DESCRIPTION
**Pull request by ckipp01**
I needed to use this a couple times in a project recently so I figured I'd try to upstream it.

### CODE REVIEW

**Review by lefou — COMMENTED** [review]
As not all `URI`s can be converted to `Path`s, this PR should also include a test case covering an error situation, which probably means to assert an thrown `IllegalArgumentException`.

File: os/src/Path.scala [hunk 1]
```diff
@@ -501,12 +504,21 @@ sealed trait PathConvertible[T] {
...
   }
   implicit object NioPathConvertible extends PathConvertible[java.nio.file.Path] {
     def apply(t: java.nio.file.Path) = t
   }
+  implicit object UriPathConvertible extends PathConvertible[URI] {
+    def apply(uri: URI) = uri.getScheme() match {
+      case "file" => Paths.get(uri)
+      case uriType =>
+        throw new IllegalArgumentException(
+          s"""os.Path can only be created from a "file" URI scheme, but found ${uriType}"""
```
Comment (lefou) [inline os/src/Path.scala:520]: I would also quote the found URI type.

**Review by lefou — COMMENTED** [review]
One minor thing below, otherwise this looks good to me.

File: os/src/Path.scala (hunk 1 above)
Comment (ckipp01) [reply os/src/Path.scala]: Sure, fixed!

**Pull request merged**